### PR TITLE
CloudStack 4.4.4 build script support

### PR DESCRIPTION
**Pull request by remibergsma**
Build a CloudStack 4.4.4 environment with XenServer and NSX to be able to test upgrades.

Uses STT for CloudStack and VXLAN for Cosmic. Will deploy all nodes with connectors for both types, so switching is easy.

The `single` NSX configs have been deleted as they don't work: a single node obviously cannot get a majority.

Anyway, when deploying you can use the `-c` flag to mention you request CloudStack instead of a Cosmic build. It'll then deploy your management server on CentOs 6 with CloudStack. Without, you get as before CentOS 7 with Cosmic.

CloudStack:
`
/data/shared/helper_scripts/cosmic/build_run_deploy.sh -m /data/shared/marvin/mct-zone1-cs1-xen1-xen2-NSX-cluster.cfg -c
`

Cosmic:
`
/data/shared/helper_scripts/cosmic/build_run_deploy.sh -m /data/shared/marvin/mct-zone1-cs1-xen1-xen2-NSX-cluster.cfg
`

### CODE REVIEW

**Comment by remibergsma** [comment]
image

image

image

**Comment by remibergsma** [comment]
Verified that KVM and NSX (with VXLAN) also still works fine:
image

File: helper_scripts/cosmic/helperlib.sh [hunk 1]
```diff
@@ -647,3 +647,22 @@ function d_show_vault_secret {
...
+  cspass=$3
+
+  # SSH/SCP helpers
+  set_ssh_base_and_scp_base ${cspass}
+
+  # Setup CloudStack management config
+  ${ssh_base} ${csuser}@${csip} "/usr/bin/cloudstack-setup-management"
+
+  # Setup CloudStack DB
+  #${ssh_base} ${csuser}@${csip} "/usr/bin/cloudstack-setup-databases cloud@localhost"
```
Comment (borisroman) [inline helper_scripts/cosmic/helperlib.sh]: This can be removed?

File: helper_scripts/cosmic/helperlib.sh [hunk 2]
```diff
@@ -647,3 +647,22 @@ function d_show_vault_secret {
...
+
+  # Start CloudStack management server
+  ${ssh_base} ${csuser}@${csip} "/etc/init.d/cloudstack-management start"
+
+  say "CloudStack management configured and started"
+}
```
Comment (borisroman) [inline helper_scripts/cosmic/helperlib.sh]: Newline

File: helper_scripts/cosmic/build_run_deploy.sh [hunk 3]
```diff
@@ -24,7 +24,8 @@ function usage {
   printf "\t-C:\tDon't use 'clean' target on maven build\n" >&2
   printf "\t-E:\tDon't use unit tests on maven build\n" >&2
   printf "\t-H:\tGit use HTTPS instead of SSH\n" >&2
-  printf "\t-S:\t(Experimental) make use of cosmic-microservices"
+  printf "\t-S:\t(Experimental) make use of cosmic-microservices\n"
+  printf "\t-c:\tDeploy CloudStack 4.4.4 artifacts instead of Cosmic build"
```
Comment (borisroman) [inline helper_scripts/cosmic/build_run_deploy.sh]: Forgot \n at the end

**Review by borisroman — CHANGES_REQUESTED** [review]

File: helper_scripts/cosmic/helperlib.sh (hunk 1 above)
Comment (remibergsma) [reply helper_scripts/cosmic/helperlib.sh]: done

File: helper_scripts/cosmic/helperlib.sh (hunk 2 above)
Comment (remibergsma) [reply helper_scripts/cosmic/helperlib.sh]: done

File: helper_scripts/cosmic/build_run_deploy.sh (hunk 3 above)
Comment (remibergsma) [reply helper_scripts/cosmic/build_run_deploy.sh]: done

**Comment by remibergsma** [comment]
@borisroman Thanks for the review, comments are addressed.

**Review by borisroman — APPROVED** [review]

**Pull request merged**